bokeh/models/axes.py

```diff
@@ -26,7 +26,7 @@ class Axis(GuideRenderer):
     Where should labels and ticks be located in relation to the axis rule.
     """)
 
-    bounds = Either(Auto, Tuple(Float, Float), help="""
+    bounds = Either(Auto, Tuple(Float, Float), Tuple(Datetime, Datetime), help="""
```

Comment:
Well, using this:
```python
import numpy as np

from bokeh.sampledata.stocks import AAPL, FB, GOOG, IBM, MSFT
from bokeh.plotting import *

output_file("stocks.html", title="stocks.py example")

left_range = np.datetime64('2000-03-01')
right_range = np.datetime64('2006-03-01')

p1 = figure(x_axis_type = "datetime")

p1.line(np.array(AAPL['date'], 'M64'), AAPL['adj_close'], color='#A6CEE3', legend='AAPL')
p1.line(np.array(FB['date'], 'M64'), FB['adj_close'], color='#1F78B4', legend='FB')
p1.line(np.array(GOOG['date'], 'M64'), GOOG['adj_close'], color='#B2DF8A', legend='GOOG')
p1.line(np.array(IBM['date'], 'M64'), IBM['adj_close'], color='#33A02C', legend='IBM')
p1.line(np.array(MSFT['date'], 'M64'), MSFT['adj_close'], color='#FB9A99', legend='MSFT')

p1.title = "Stock Closing Prices"
p1.grid.grid_line_alpha=0.3
p1.xaxis.axis_label = 'Date'
p1.yaxis.axis_label = 'Price'
p1.xaxis.bounds = (left_range, right_range)

show(p1)  # open a browser
```

I get this plot...

![bounds](https://cloud.githubusercontent.com/assets/1640669/6158774/a3ad3104-b22c-11e4-8831-68e74ce35a91.png)

Seems to work...